extend_all.sh

```diff
@@ -1,14 +1,20 @@
 #!/bin/bash
 . util.sh
 
-if [[ $# -ne 2 ]] || [[ ! -e "$1" ]] || [[ ! -e "$2" ]] ; then
-	echo "Usage: $0 path/to/bzImage path/to/initrd"
-	exit
+# see https://www.kernel.org/doc/html/latest/x86/boot.html#details-of-header-fields
+if [[ -e "$1" ]] ; then
+	KERNEL_PROT_SKIP=$((`hexdump "$1" -s0x1f1 -n1 -e '/1 "%u"'` * 512 + 512))
 fi
 
-# see https://www.kernel.org/doc/html/latest/x86/boot.html#details-of-harder-fileds
-KERNEL_PROT_SKIP=$((`hexdump "$1" -s0x1f1 -n1 -e '/1 "%u"'` * 512 + 512))
+if [[ $# -eq 2 ]] && [[ -e "$1" ]] && [[ -e "$2" ]] ; then
+	extend_sha1 "$(extend_sha1)" "`dd if="$1" bs=1 skip=$KERNEL_PROT_SKIP 2>/dev/null | sha1sum`" "$2"
```

Comment:
I think this is logically correct but for me I find it easier to read/reason if the inner `extend_shaXXX` and its parameter is under the same subshell call, i.e.
```
extend_sha1 "$(extend_sha1 `dd if="$1" bs=1 skip=$KERNEL_PROT_SKIP 2>/dev/null | sha1sum`)" "$2"
```